pg2mysql.pl

```diff
@@ -96,63 +99,80 @@ sub handle_line {
 
     # Explicitly die when we encounter lines we can't handle
     if ( $strict ) {
-	if ( $line =~ /CREATE TYPE / ) {
-	    die "CREATE TYPE statements not supported";
-	}
+        if ( $line =~ /CREATE TYPE / ) {
+            die "CREATE TYPE statements not supported";
+        }
     }
-    
+
     # Explicitly skipped statments need to be defined first.
     if ( $in_begin_end || $line =~ m/^\s*begin\s*$/i) {
         ($line, $in_begin_end, $skip) = handle_begin_end($line);
     } elsif ( $line =~ m/pg_catalog\.setval/ ) {
-	$line = handle_setval($line);
+        $line = handle_setval($line);
     } elsif ( $in_create || $line =~ m/^\s*CREATE TABLE/ ) {
         ($line, $in_create, $skip) = handle_create($line);
     } elsif ( $in_alter || $line =~ m/^\s*ALTER TABLE/ ) {
         ($line, $in_alter, $skip) = handle_alter($line, $nextline);
     } elsif ( $in_insert || $line =~ m/^\s*INSERT INTO/ ) {
         ($line, $in_insert, $skip) = handle_insert($line, $nextline);
+    } elsif ( $in_copy || $line =~ m/^\s*COPY/ ) {
+        ($line, $in_copy, $skip) = handle_copy($line, $nextline);
     } elsif ( $line =~ m/^\s*CREATE (UNIQUE )?INDEX/ ) {
         ($line, $skip) = handle_create_index($line);
     } else {
         print_warning("$line");
         return;
     }
 
+    # Very special case: key is a reserved word in MySQL, rename all
+    $line =~ s/`key`/`index`/g;
+
     print "$line\n" unless $skip;
 }
 
 sub handle_create {
     my $line = shift;
+    # special column names that can be replaced by a MySQL keyword
+    my @special = ('serial', 'uuid', 'text', 'character');
+
+    # temp rename special columns
+    foreach my $col ( @special ) {
+        $line =~ s/^\s{4}$col/    $col\_/g;
```

Comment:
This seems like a *very* specific regex matching only these names after 4 spaces on a newline. Can we do something a bit more general? Even `^\s+` would probably be better. 

And why append an `_` to the column? I think this could use a better comment at least.

Reply:
This is a way to prevent a column name being wrongly replaced by a column instruction (regarding #50)

I know this seems a bit weird at first but thanks to that, the specific colums are not messed up. (i.e. character column that can be inside a play table, or the serial column that can be an activation key in a table)

Reply:
Actually, my case was 4 spaces but it could be \s+ yup, I will commit it on monday. For the _, it's just a temporary symbol, probably we can place it as a prefix instead of a suffix :)

Reply:
I see. This is to prevent columns from being renamed.